### ISSUE TITLE
MIBs dynamically read or static?

### ISSUE THREAD

**Issue by carlklatsky** [rlaager/docsis#76]
Hi,

Not sure if this is still an active project, but wanted to know if the MIB files in the ./mibs directory are dynamically read at file decode time or statically linked at compile time?  I had thought they were dynamically read when the command is issued to decode a DOCSIS file, but it seems like it is not working for me.  I am sure the MIB file to decode an SNMP OID into its name is in the ./mibs directory, but the decoded file still listed it by OID.

	SnmpMibObject .1.3.6.1.2.1.69.1.2.1.2.1 IPAddress 0.0.0.0;
	SnmpMibObject .1.3.6.1.2.1.69.1.2.1.3.1 IPAddress 0.0.0.0;

Do files added to the ./mibs directory need a specific file extension to be read in?

Thanks,
Carl

**Comment by rlaager** [comment]
The project is more or less inactive. I try to look at things occasionally, but I don't even use the utility myself.

**Comment by carlklatsky** [comment]
Understood.  I will dig into it some more when I have time, but marking this issue closed for now.  Thanks.

**Comment by carlklatsky** [comment]
Just an addendum on my findings.  I am building under cygwin environment.  When running autogen.sh, which I think also runs configure.sh, it seems like the mib path variable NETSNMP_MIBPATH in the Makefile in the main build directory is not getting set.  I manually edited that variable to be NETSNMP_MIBPATH=$(CYGWIN_BUILDROOT)/mibs, which is where I added additional mib files.  Then re-build which incorporates the MIBs in that file, then decode the file and this time the OIDs that were not decoded to name now are decoded to name.

And I think this answers my own question that the MIB files get linked statically at build time.  New MIB files cannot just be dropped in the /mibs directory and expect to be available for decode.
